flytekit/core/promise.py

```diff
@@ -793,7 +793,77 @@ def create_and_link_node(
     for output_name, output_var_model in typed_interface.outputs.items():
         # TODO: If node id gets updated later, we have to make sure to update the NodeOutput model's ID, which
         #  is currently just a static str
-        node_outputs.append(Promise(output_name, NodeOutput(node=non_sdk_node, var=output_name)))
+        node_outputs.append(Promise(output_name, NodeOutput(node=flytekit_node, var=output_name)))
         # Don't print this, it'll crash cuz sdk_node._upstream_node_ids might be None, but idl code will break
 
     return create_task_output(node_outputs, interface)
+
+
+class LocallyExecutable(Protocol):
+    def _local_execute(self, ctx: FlyteContext, **kwargs) -> Union[Tuple[Promise], Promise, VoidPromise]:
+        ...
+
+
+def executable_artifact_call_handler(entity: Union[SupportsNodeCreation, LocallyExecutable], *args, **kwargs):
+    """
```

Comment:
the logic is the same for workflows and tasks, but we did not want to create base class, hence a separate method